pkg/service/storage/service.go

```diff
@@ -258,7 +259,14 @@ func New(opts ...Option) (*StorageService, error) {
 		return nil, fmt.Errorf("creating claim service: %w", err)
 	}
 
-	repl, err := replicator.New(id, pdpImpl, blobs, claims, receiptStore, uploadServiceConnection)
+	if c.replicatorDB == nil {
+		c.replicatorDB, err = sqlitedb.NewMemory()
+		if err != nil {
+			return nil, fmt.Errorf("creating in-memory replicator db: %w", err)
+		}
+	}
+
+	repl, err := replicator.New(id, pdpImpl, blobs, claims, receiptStore, uploadServiceConnection, c.replicatorDB)
```

Comment:
I'd prefer to make the DB a required configuration for the replicator, but unsure how to make that work with current config design. I could fail here if one isn't provided, but that will break other services that use this constructor, such as `pkg/aws/service.go` which are mostly stateless. On principle, the aws service implementation ought to use some aws specific cloud db, but that's beyond the scope of this implementation.

I am assuming the aws package will be removed at some point, since I think it's mostly for testing anyways? At which point we can re-think how these services are instantiated.